grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -48,21 +48,29 @@ import org.pih.warehouse.LocalizationUtil
  */
 class Product implements Comparable, Serializable {
 
-    def beforeInsert = {
+    def beforeInsert() {
         createdBy = AuthService.currentUser
```

Comment:
In the Shipment domain I changed the convention to include updatedBy in the insert as well. I had assumed we were already doing this everywhere else but it looks like I was wrong.

Topics for tech-huddle maybe: 
- Do you think that should be applied everywhere? 
- If so do you think we can move this to a generic solution 
- Auditable abstract class or trait (assuming that's what traits are for) that adds these methods to domains that have these audit fields?

The first of these could be done during MVP1, but the other points should be MVP2/post-migration.

Now that we might have more consistency with the Grails 3 dirty properties API we could also try to make these persistence events more fine-grained. There are only a few properties on Product that should cause a refresh to InventorySnapshot. Same with other domains for ProductAvailability.